src/content/en/2019/javascript.md

```diff
@@ -0,0 +1,338 @@
+---
+part_number: II
+chapter_number: 1
+title: JavaScript
+authors: [housseindjirdeh]
+reviewers: [rviscomi, obto, paulcalvano, mathiasbynens]
+---
+
+# Chapter 1: JavaScript
+
+JavaScript is a scripting language that makes it possible to build interactive and complex
+experiences on the web. This includes responding to user interactions, updating dynamic content on a
+page and so forth. Anything involving how a web page should behave when an event occurs is what
+JavaScript is used for.
+
+The language specification itself, along with many community-built libraries and frameworks used by
+developers around the world, has changed and evolved ever since the language was created in 1995.
+JavaScript implementations and interpreters have also continued to progress, making the language
+usable in many environments, not only web browsers.
+
+The [HTTP Archive](https://httparchive.org/) crawls [millions of
+URLs](https://httparchive.org/reports/state-of-the-web#numUrls) every month and runs them through a
+private instance of [WebPageTest](https://webpagetest.org/) to store key information of every page
+(see [methodology](/methodology)). In the context of JavaScript, HTTP Archive provides extensive
+information on the usage of the language for the entire web. This chapter consolidates and analyzes
+many of these trends. 
+
+# How much JavaScript do we use?
+
+JavaScript is the most costly resource we send to browsers, having to be downloaded, parsed,
+compiled and then finally executed. Although browsers have significantly decreased the time it takes
+to parse and compile scripts,
+[download and execution have become the most expensive stages](https://v8.dev/blog/cost-of-javascript-2019)
+when JavaScript is processed by a web page.
+
+Sending smaller JavaScript bundles to the browser is the best way to reduce download times, and in
+turn improve page performance. **But how much JavaScript do we really use?**
+
+Here's a chart showing the amount of bytes shipped for different percentiles:
+
+<insert graphic of metric 01_01a>
```

Comment:
@rviscomi Not sure how we should insert charts in the article. Is there a preferred way? Screenshots okay?

Wanted to leave it out for now since I assume we're going to do a pass making sure all the visualizations are consist color/design-wise.

Reply:
Placeholders like this would be perfect, thanks.